### ISSUE TITLE
Web Components

### ISSUE THREAD

**Issue by Boscop** [tcr/rust-webplatform#19]
Web Components are the future. It would be nice if one could build web components in Rust.
For those that don't know, web components allow you to create custom elements that you can reuse (e.g. [this element](https://github.com/PolymerElements/marked-element) turns it's argument from markdown into html), it makes web dev more modular and it's native without requiring a framework.
Here some links about the current browser support for web components:
https://jonrimmer.github.io/are-we-componentized-yet/
https://www.polymer-project.org/1.0/docs/browsers (Polymer provides polyfills to make web components available on all browsers even if they don't support web components yet.)
https://vaadin.com/blog/-/blogs/web-components-in-production-use-are-we-there-yet-

The new youtube website uses Polymer (you can see it now already by setting a cookie), also big brands like Comcast, Net-a-Porter, etc. have been using Polymer / web components for some time. (You can watch dev talks from those companies on youtube.)
[Over 4 million websites in the wild are using Polymer / Web Components](https://youtu.be/fFF2Yup2dMM?t=953)

TL;DR: Web Components are the future of the web, Rust is the future anyway, so:
Rust + Web Components = WIN

**Comment by davidhewitt** [comment]
@Boscop I actually have a hacked-up toy project which allowed me to do just that.

I could submit it to this crate in a pull request?

**Comment by Boscop** [comment]
Yes, it would be interesting.

Are you also generating the html and css code, or are you expressing the web components as pure js, with html and css as js string?

**Comment by danpaz** [comment]
@davidhewitt do you have that project up anywhere?

**Comment by davidhewitt** [comment]
Afraid I don't; the code doesn't want to compile for me any more anyway, sorry (something with emscripten I think). I've been pretty busy as of late but actually over the christmas break I'm hoping to revisit the code and get it running with the wasm32-unknown-unknown target so I don't have to try and figure out how I broke emscripten. I'll be sure to post back here when I get something going!
